File: Números combinatorios/Python/numero_combinatorio_v1.py
def factorial(n):
    productoria = 1
    for i in range(2, n+1):
        productoria *= i
    return productoria

def combinatorio(n, k):
    return factorial(n)//(factorial(n-k)*factorial(k))

n=int(input("Ingrese n: "))
k=int(input("Ingrese k: "))

if n <= 0 or k <= 0:
    print("n y k deben ser mayores que cero")
elif k > n:
    print("n debe ser mayor o igual que k")
else:
    print("El combinatorio de n en k es {:d}".format(combinatorio(n, k)))

File: Números combinatorios/Java/Numero_Combinatorio_v1.java
package numero_combinatorio_v1;

import java.util.Scanner;

public class Numero_Combinatorio_v1 {

    public static long factorial(int n) {
        long productoria = 1;
        for (int i = 2; i <= n; i++) {
            productoria *= i;
        }
        return productoria;
    }

    public static long combinatorio(int n, int k) {
        return factorial(n) / (factorial(n - k) * factorial(k));
    }

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        System.out.printf("Ingrese n: ");
        int n = reader.nextInt();
        System.out.printf("Ingrese k: ");
        int k = reader.nextInt();

        if (n <= 0 || k <= 0) {
            System.out.printf("n y k deben ser mayores que cero%n");
        } else if (k > n) {
            System.out.printf("n debe ser mayor o igual que k%n");
        } else {
            System.out.printf("El combinatorio de n en k es %d%n", combinatorio(n, k));
        }
    }
}
